vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpContext.java

```diff
@@ -293,4 +301,18 @@ public HttpContext set(String key, Object value) {
     }
     return this;
   }
+
+  public HttpContext onClientRequest(Handler<HttpClientRequest> interceptor) {
+    if (interceptor != null) {
+      requestInterceptors.add(interceptor);
+    }
+    return this;
+  }
+
+  public HttpContext onClientResponse(Handler<HttpClientResponse> interceptor) {
```

Comment:
this should be clarified in javadoc as to when it gets set off. For tracing, would be nice for this to be set after headers were read, but not after body is consumed

Reply:
I think we should add a test that mimicks tracing